arangod/Cluster/DropCollection.cpp

```diff
@@ -64,11 +81,15 @@ bool DropCollection::first() {
     });
 
   if (found.fail()) {
-    std::string errorMsg("DropCollection: Failed to lookup local collection ");
-    errorMsg += collection + "in database " + database;
+    std::string errorMsg("DropCollection: failed to lookup local collection ");
+    errorMsg += database + "/" + collection;
     _result.reset(TRI_ERROR_ARANGO_DATABASE_NOT_FOUND, errorMsg);
+    LOG_TOPIC(ERR, Logger::MAINTENANCE) << errorMsg;
+    setState(FAILED);
```

Comment:
FYI:  MaintenanceWorker::nextState() will also setState(FAILED) or setState(COMPLETE) based upon _result.  Hmm, which implies that the new fail() and complete() functions are redundant everywhere, except for fail() in Action constructors.